yt/yt/server/master/node_tracker_server/node_tracker.cpp

```diff
@@ -1053,18 +1055,47 @@ class TNodeTracker
             }
         }
 
-        TRack* oldNodeRack = nullptr;
+        TRack* rack = nullptr;
 
         auto* node = FindNodeByAddress(options.DefaultAddress);
         auto isNodeNew = !IsObjectAlive(node);
         if (!isNodeNew) {
             KickOutPreviousNodeIncarnation(node, options.DefaultAddress);
-            oldNodeRack = node->GetRack();
+            rack = node->GetRack();
+        }
+        if (options.Rack.has_value()) {
```

Comment:
We tend to use implicit casts for std::optionals, so just `if (options.Rack)` here